kratos/utilities/scope_lock.cpp

```diff
@@ -0,0 +1,54 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher, https://github.com/philbucher
+//                   Suneth Warnakulasuriya, https://github.com/sunethwarna
+//
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "utilities/scope_lock.h"
+
+
+namespace Kratos {
+
+ScopeLock::ScopeLock()
+{
+    KRATOS_ERROR_IF_NOT(ScopeLock::mIsInitialized) << "ScopeLock is not initialized!" << std::endl;
+    mpLock->SetLock();
+}
+
+ScopeLock::~ScopeLock()
+{
+    mpLock->UnSetLock();
+}
+
+bool ScopeLock::IsInitialized()
+{
+    return mIsInitialized;
+}
+
+void ScopeLock::Initialize()
+{
+    KRATOS_ERROR_IF(ScopeLock::mIsInitialized) << "ScopeLock was already initialized!" << std::endl;
+
+    static LockObject static_lock;
+    mpLock = &static_lock;
```

Comment:
are u sure this is kept alive this way? mplock is not a shred pointer, so i think this dies as soon as you go out of the scope...

Reply:
thats why I made it static
I used the same mechanism as the [`GiDBaseIO` does](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/sources/gid_io.cpp) (at least I think so)